src/DotNetWorker.Core/Context/RetryContext.cs

```diff
@@ -12,12 +12,12 @@ public abstract class RetryContext
         /// <summary>
         /// The the retry count for the current event.
         /// </summary>
-        public abstract int RetryCount { get; }
+        public abstract int? RetryCount { get; }
```

Comment:
Unfortunately, this is a breaking change (as well as the `MaxRetryCount` change) - the type is going from `int` to `Nullable<int>`. We will need to find a way to solve this without breaking changes to the public API surface.

https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules

Reply:
I changed the value to be -1 if the RetryContext is null.